### PR TITLE
Add copy button to document pages 

### PR DESCRIPTION
**Pull request by vietluong2110**

### CODE REVIEW

**Comment by davidteather** [comment]
looks like the clipmark gets a little odd on small screens
image


here's the css styles I used on my personal site, which might help
```css

.prose pre code {
    display: block; 
    white-space: pre-wrap;
    word-break: break-word;
    padding: 0;
    background-color: inherit;
    border-radius: inherit;
    color: #adbac7;
}


span[data-rehype-pretty-code-title] + pre {
    margin-top: -0.2rem;
    border-top-left-radius: 0;
}

span[data-rehype-pretty-code-title] {
    background-color: #22272e;
    color: #adbac7;
    padding: 4px;
    padding-left: 8px;
    padding-right: 8px;
    border-radius: 4px;
    border-bottom-right-radius: 0;
    border-bottom-left-radius: 0;
    font-size: 0.8rem;
}

.highlighted {
    background-color: #57c7f933;
}
code {
    background-color: #22272e !important;
    color: #adbac7 !important;
    font-size: 0.9rem !important;
}
```

and had this rehype plugin since I was running into issues with <figure>
```tsx

function rehypeReplaceFigureAndCaption() {
    return (tree) => {
        visit(tree, 'element', (node, index, parent) => {
            // Change <figure> to <div> if it has the specific data attribute
            if (node.tagName === 'figure' && node.properties && node.properties['data-rehype-pretty-code-figure'] !== undefined) {
                node.tagName = 'div';  // Change the tag name to <div>
            }
            // Change <figcaption> to <span>
            if (node.tagName === 'figcaption' && node.properties && node.properties['data-rehype-pretty-code-title'] !== undefined) {
                node.tagName = 'span';
            }
        });
    };
}
```

and for extra functionality allowing highlighting of nodes
```js
[rehypePrettyCode, {
            transformers: [
                transformerCopyButton({
                    visibility: 'always',
                    feedbackDuration: 500,
                }),
            ],
            onVisitLine(node) {
                // Prevent lines from collapsing in `display: grid` mode, and
                // allow empty lines to be copy/pasted
                if (node.children.length === 0) {
                    node.children = [{ type: 'text', value: ' ' }];
                }
            },
            onVisitHighlightedLine(node) {
                if (!node.properties.className) {
                    node.properties.className = [];
                }
                node.properties.className.push('highlighted');
            },
        }]
```

would be nice to support like

> \`\`\`js title="mypagename.tsx"
the above code should be able to do this along with highlughting specific lines within the block as it does in my portfolio website like
> \`\`\`astro title="components/AstroSearch.astro" {48-53, 57, 63-70}

File: package-lock.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (davidteather) [inline package-lock.json:1]: delete this file, no npm on the root level only within `website`

File: website/lib/markdown.ts [hunk 2]
```diff
@@ -84,7 +86,15 @@ export async function processMarkdown(
         });
       };
     })
-    //.use(addCopyButtonToCodeBlocks)
+    .use(rehypePrettyCode, {
+      transformers: [
+        transformerCopyButton({
+          visibility: 'always',
+          feedbackDuration: 3_000,
```
Comment (davidteather) [inline website/lib/markdown.ts:93]: drop to `500` for feedback duration

**Review by davidteather — CHANGES_REQUESTED** [review]

**Comment by vietluong2110** [comment]
@davidteather  I adjusted the position of the copy button. Thank you for your css code snippet!

File: .gitignore [hunk 3]
```diff
@@ -1,3 +1,4 @@
 screenshot.py
 *.DS_Store
-*.tfvars
\ No newline at end of file
+*.tfvars
+*.json
```
Comment (davidteather) [inline .gitignore:4]: why adding `*.json`?

**Review by davidteather — CHANGES_REQUESTED** [review]
* why json in `gitignore`

otherwise looks look, there's some overflow with single tick code blocks but I'll try to deal with that later

**Review by davidteather — APPROVED** [review]

**Pull request merged**